src/Microsoft.DncEng.SecretManager/SecretTypes/GitHubBotAccount.cs

```diff
@@ -0,0 +1,205 @@
+            }
+            else
+            {
+                string recoveryCodes = await context.GetSecretValue(context.SecretName + RecoveryCodesSuffix);
+                secrets.Add(new SecretData(recoveryCodes, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue));
+            }
+
+            if (rollSecret)
+            {
+                var newSecret = await _console.PromptAsync("Enter the new sescret: ");
+                seed = ConvertFromBase32(newSecret);
+                await ShowOneTimePassword(seed);
+                secrets.Add(new SecretData(newSecret, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue));
+            }
+            else
+            {
+                secrets.Add(new SecretData(secret, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue));
+            }
+
+            return secrets;
+        }
+
+        private async Task<List<SecretData>> NewAccount(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            _console.WriteLine($"Please sign up for a new GitHub account {parameters.Name}.");
+
+            var password = PasswordGenerator.GenerateRandomPassword(15);
+            var customPassword = await _console.PromptAsync($"Press enter to use generated password {password} or enter a new one: ");
+            if (!string.IsNullOrWhiteSpace(customPassword))
+                password = customPassword.Trim();
+
+            _console.WriteLine("Enable two factor authentification using Authenticator app.");
+
+            var recoveryCodes = await _console.PromptAsync("Enter recovery codes: ");
+            var secret = await _console.PromptAsync("Enter secret: ");
+            var seed = ConvertFromBase32(secret);
+
+            await ShowOneTimePassword(seed);
+
+            return new List<SecretData> {
+                new SecretData(password, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue),
+                new SecretData(recoveryCodes, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue),
+                new SecretData(secret, DateTimeOffset.MaxValue, DateTimeOffset.MaxValue)};
+        }
+
+        private async Task ShowOneTimePassword(byte[] seed)
+        {
+            var generateTotp = true;
+            while (generateTotp)
+            {
+                var oneTimePassword = GenerateOneTimePassword(_clock.UtcNow, seed);
+                generateTotp = await _console.ConfirmAsync($"Your one time password: {oneTimePassword}. Enter yes to generate another one: ");
+            }
+        }
+
+        private static string GenerateOneTimePassword(DateTimeOffset timestamp, byte[] seed)
+        {
+            DateTimeOffset dateTimeOffset = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+            TimeSpan timeSpan = TimeSpan.FromSeconds(30.0);
+            byte[] bytes = BitConverter.GetBytes((long)((timestamp - dateTimeOffset).TotalMilliseconds / timeSpan.TotalMilliseconds));
```

Comment:
IIRC there are some ToUnixTime methods on DateTimeOffset, you should be able to use those rather than subtract from 1970.

Reply:
Definitely you are right, it's much better with ToUnixTimeSeconds.